pkg/openshift/template.go

```diff
@@ -0,0 +1,127 @@
+package openshift
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/maistra/istio-workspace/pkg/assets"
+
+	openshiftApi "github.com/openshift/api/template/v1"
+
+	"gopkg.in/yaml.v2"
+
+	apiextv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+var re = regexp.MustCompile(`(?m)\${(.*?)}`)
+
+const substitution = "{{ .$1 }}"
+
+// ProcessTemplateUsingEnvVars takes template path, loads its content and
+// substitutes variables by using those which are defined as environment variables.
+// Returns processed template as byte array
+func ProcessTemplateUsingEnvVars(templatePath string) ([]byte, error) {
```

Comment:
Would have been nice if maybe as part of 'ike install-operator --help' it displayed which ENV variables were available to manipulate. A little dump of the Template.parameters. (note: there is a "description" field on the parameters as well we could use=

Reply:
Yeah, I'm working on it as part of #142